### ISSUE TITLE
[platform-linux][shared_preferences] Exception throws in `remove` method: Unhandled Exception: type 'Null' is not a subtype of type 'Object' in type cast

### ISSUE THREAD

**Issue by nghialm269** [flutter/flutter#78404]
## Steps to Reproduce

1. Run 
```sh
flutter create bug
cd bug
dart migrate --apply-changes
flutter pub add shared_preferences
```
2. Update the files as follows:
* Edit `lib/main.dart`:

```dart
import 'package:shared_preferences/shared_preferences.dart';

void main() async {
  final prefs = await SharedPreferences.getInstance();
  await prefs.remove("test");
}
```

**Expected results:**

No exception throws.

**Actual results:** 
```
[ERROR:flutter/lib/ui/ui_dart_state.cc(186)] Unhandled Exception: type 'Null' is not a subtype of type
 'Object' in type cast
#0      CastMap.remove (dart:_internal/cast.dart:280:48)
#1      SharedPreferencesLinux.remove (package:shared_preferences_linux/shared_preferences_linux.dart:
92:17)
<asynchronous suspension>
#2      main (package:bug/main.dart:7:3)
<asynchronous suspension>
```

<details>
  <summary>Logs</summary>

```
[  +61 ms] executing: [/home/nghialm/.asdf/installs/flutter/2.0.2-stable/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[  +28 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] 8962f6dc68ec8e2206ac2fa874da4a453856c7d3
[        ] executing: [/home/nghialm/.asdf/installs/flutter/2.0.2-stable/] git tag --points-at 8962f6dc68ec8e2206ac2fa874da4a453856c7d3
[  +10 ms] Exit code 0 from: git tag --points-at 8962f6dc68ec8e2206ac2fa874da4a453856c7d3
[        ] 2.0.2
[  +31 ms] executing: [/home/nghialm/.asdf/installs/flutter/2.0.2-stable/] git rev-parse --abbrev-ref --symbolic @{u}
[   +4 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/stable
[        ] executing: [/home/nghialm/.asdf/installs/flutter/2.0.2-stable/] git ls-remote --get-url origin
[   +4 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[  +32 ms] executing: [/home/nghialm/.asdf/installs/flutter/2.0.2-stable/] git rev-parse --abbrev-ref HEAD
[   +5 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] stable
[  +37 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +1 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +54 ms] executing: /opt/android-sdk/platform-tools/adb devices -l
[  +34 ms] List of devices attached
[   +2 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +70 ms] Skipping pub get: version match.
[  +23 ms] Found plugin path_provider_linux at /home/nghialm/.asdf/installs/flutter/2.0.2-stable/.pub-cache/hosted/pub.dartlang.org/path_provider_linux-2.0.0/
[   +5 ms] Found plugin path_provider_windows at /home/nghialm/.asdf/installs/flutter/2.0.2-stable/.pub-cache/hosted/pub.dartlang.org/path_provider_windows-2.0.0/
[   +4 ms] Found plugin shared_preferences at /home/nghialm/.asdf/installs/flutter/2.0.2-stable/.pub-cache/hosted/pub.dartlang.org/shared_preferences-2.0.5/
[   +2 ms] Found plugin shared_preferences_linux at /home/nghialm/.asdf/installs/flutter/2.0.2-stable/.pub-cache/hosted/pub.dartlang.org/shared_preferences_linux-2.0.0/
[   +1 ms] Found plugin shared_preferences_macos at /home/nghialm/.asdf/installs/flutter/2.0.2-stable/.pub-cache/hosted/pub.dartlang.org/shared_preferences_macos-2.0.0/
[   +2 ms] Found plugin shared_preferences_web at /home/nghialm/.asdf/installs/flutter/2.0.2-stable/.pub-cache/hosted/pub.dartlang.org/shared_preferences_web-2.0.0/
[   +1 ms] Found plugin shared_preferences_windows at /home/nghialm/.asdf/installs/flutter/2.0.2-stable/.pub-cache/hosted/pub.dartlang.org/shared_preferences_windows-2.0.0/
[  +78 ms] Found plugin path_provider_linux at /home/nghialm/.asdf/installs/flutter/2.0.2-stable/.pub-cache/hosted/pub.dartlang.org/path_provider_linux-2.0.0/
[   +1 ms] Found plugin path_provider_windows at /home/nghialm/.asdf/installs/flutter/2.0.2-stable/.pub-cache/hosted/pub.dartlang.org/path_provider_windows-2.0.0/
[   +2 ms] Found plugin shared_preferences at /home/nghialm/.asdf/installs/flutter/2.0.2-stable/.pub-cache/hosted/pub.dartlang.org/shared_preferences-2.0.5/
[   +1 ms] Found plugin shared_preferences_linux at /home/nghialm/.asdf/installs/flutter/2.0.2-stable/.pub-cache/hosted/pub.dartlang.org/shared_preferences_linux-2.0.0/
[        ] Found plugin shared_preferences_macos at /home/nghialm/.asdf/installs/flutter/2.0.2-stable/.pub-cache/hosted/pub.dartlang.org/shared_preferences_macos-2.0.0/
[   +1 ms] Found plugin shared_preferences_web at /home/nghialm/.asdf/installs/flutter/2.0.2-stable/.pub-cache/hosted/pub.dartlang.org/shared_preferences_web-2.0.0/
[   +1 ms] Found plugin shared_preferences_windows at /home/nghialm/.asdf/installs/flutter/2.0.2-stable/.pub-cache/hosted/pub.dartlang.org/shared_preferences_windows-2.0.0/
[  +14 ms] Generating /home/nghialm/learn/flutter/bug/android/app/src/main/java/io/flutter/plugins/GeneratedPluginRegistrant.java
[  +63 ms] Initializing file store
[   +6 ms] Skipping target: gen_localizations
[   +3 ms] complete
[   +4 ms] Launching lib/main.dart on Linux in debug mode...
[   +4 ms] /home/nghialm/.asdf/installs/flutter/2.0.2-stable/bin/cache/dart-sdk/bin/dart --disable-dart-dev /home/nghialm/.asdf/installs/flutter/2.0.2-stable/bin/cache/artifacts/engine/linux-x64/frontend_server.dart.snapshot --sdk-root /home/nghialm/.asdf/installs/flutter/2.0.2-stable/bin/cache/artifacts/engine/common/flutter_patched_sdk/ --incremental --target=flutter --debugger-module-names --experimental-emit-debug-metadata --output-dill /tmp/flutter_tools.XLCWMU/flutter_tool.ILDLVI/app.dill --packages /home/nghialm/learn/flutter/bug/.dart_tool/package_config.json -Ddart.vm.profile=false -Ddart.vm.product=false --enable-asserts --track-widget-creation --filesystem-scheme org-dartlang-root --initialize-from-dill build/cache.dill.track.dill
[  +13 ms] Building Linux application...
[   +9 ms] <- compile package:bug/main.dart
[   +3 ms] executing: [build/linux/debug/] cmake -G Ninja -DCMAKE_BUILD_TYPE=Debug /home/nghialm/learn/flutter/bug/linux
[ +165 ms] -- The CXX compiler identification is Clang 11.1.0
[   +5 ms] -- Detecting CXX compiler ABI info
[ +168 ms] -- Detecting CXX compiler ABI info - done
[   +7 ms] -- Check for working CXX compiler: /usr/bin/clang++ - skipped
[        ] -- Detecting CXX compile features
[        ] -- Detecting CXX compile features - done
[   +4 ms] -- Found PkgConfig: /usr/bin/pkg-config (found version "1.7.3") 
[        ] -- Checking for module 'gtk+-3.0'
[  +11 ms] --   Found gtk+-3.0, version 3.24.27
[ +268 ms] -- Checking for module 'glib-2.0'
[   +7 ms] --   Found glib-2.0, version 2.66.7
[  +20 ms] -- Checking for module 'gio-2.0'
[   +9 ms] --   Found gio-2.0, version 2.66.7
[  +25 ms] -- Checking for module 'blkid'
[   +7 ms] --   Found blkid, version 2.36.2
[  +18 ms] -- Checking for module 'liblzma'
[   +8 ms] --   Found liblzma, version 5.2.5
[  +23 ms] -- Configuring done
[   +6 ms] -- Generating done
[   +4 ms] -- Build files have been written to: /home/nghialm/learn/flutter/bug/build/linux/debug
[   +5 ms] executing: ninja -C build/linux/debug install
[   +7 ms] ninja: Entering directory `build/linux/debug'
[+15253 ms] [1/6] Generating /home/nghialm/learn/flutter/bug/linux/flutter/ephemeral/libflutter_linux_gtk.so, /home/nghialm/learn/flutter/bug/linux/flutter/ephemeral/flutter_linux/fl_basic_message_channel.h, /home/nghialm/learn/flutter/bug/linux/flutter/ephemeral/flutter_linux/fl_binary_codec.h, /home/nghialm/learn/flutter/bug/linux/flutter/ephemeral/flutter_linux/fl_binary_messenger.h, /home/nghialm/learn/flutter/bug/linux/flutter/ephemeral/flutter_linux/fl_dart_project.h, /home/nghialm/learn/flutter/bug/linux/flutter/ephemeral/flutter_linux/fl_engine.h, /home/nghialm/learn/flutter/bug/linux/flutter/ephemeral/flutter_linux/fl_json_message_codec.h, /home/nghialm/learn/flutter/bug/linux/flutter/ephemeral/flutter_linux/fl_json_method_codec.h, /home/nghialm/learn/flutter/bug/linux/flutter/ephemeral/flutter_linux/fl_message_codec.h, /home/nghialm/learn/flutter/bug/linux/flutter/ephemeral/flutter_linux/fl_method_call.h, /home/nghialm/learn/flutter/bug/linux/flutter/ephemeral/flutter_linux/fl_method_channel.h, /home/nghialm/learn/flutter/bug/linux/flutter/ephemeral/flutter_linux/fl_method_codec.h, /home/nghialm/learn/flutter/bug/linux/flutter/ephemeral/flutter_linux/fl_method_response.h, /home/nghialm/learn/flutter/bug/linux/flutter/ephemeral/flutter_linux/fl_plugin_registrar.h, /home/nghialm/learn/flutter/bug/linux/flutter/ephemeral/flutter_linux/fl_plugin_registry.h, /home/nghialm/learn/flutter/bug/linux/flutter/ephemeral/flutter_linux/fl_standard_message_codec.h, /home/nghialm/learn/flutter/bug/linux/flutter/ephemeral/flutter_linux/fl_standard_method_codec.h, /home/nghialm/learn/flutter/bug/linux/flutter/ephemeral/flutter_linux/fl_string_codec.h, /home/nghialm/learn/flutter/bug/linux/flutter/ephemeral/flutter_linux/fl_value.h, /home/nghialm/learn/flutter/bug/linux/flutter/ephemeral/flutter_linux/fl_view.h, /home/nghialm/learn/flutter/bug/linux/flutter/ephemeral/flutter_linux/flutter_linux.h, _phony_
[        ] [  +95 ms] executing: [/home/nghialm/.asdf/installs/flutter/2.0.2-stable/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] [  +37 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] [        ] 8962f6dc68ec8e2206ac2fa874da4a453856c7d3
[        ] [        ] executing: [/home/nghialm/.asdf/installs/flutter/2.0.2-stable/] git tag --points-at 8962f6dc68ec8e2206ac2fa874da4a453856c7d3
[        ] [  +13 ms] Exit code 0 from: git tag --points-at 8962f6dc68ec8e2206ac2fa874da4a453856c7d3
[        ] [        ] 2.0.2
[        ] [  +44 ms] executing: [/home/nghialm/.asdf/installs/flutter/2.0.2-stable/] git rev-parse --abbrev-ref --symbolic @{u}
[        ] [   +5 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] [        ] origin/stable
[        ] [        ] executing: [/home/nghialm/.asdf/installs/flutter/2.0.2-stable/] git ls-remote --get-url origin
[        ] [   +5 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] [        ] https://github.com/flutter/flutter.git
[        ] [  +43 ms] executing: [/home/nghialm/.asdf/installs/flutter/2.0.2-stable/] git rev-parse --abbrev-ref HEAD
[        ] [   +5 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] [        ] stable
[        ] [  +49 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] [   +2 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[        ] [  +56 ms] Artifact Instance of 'MaterialFonts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'GradleWrapper' is not required, skipping update.
[        ] [        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterSdk' is not required, skipping update.
[        ] [        ] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FontSubsetArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'PubDependencies' is not required, skipping update.
[        ] [ +130 ms] Initializing file store
[        ] [  +17 ms] Skipping target: gen_localizations
[        ] [   +6 ms] unpack_linux: Starting due to {}
[        ] [ +108 ms] kernel_snapshot: Starting due to {}
[        ] [   +4 ms] unpack_linux: Complete
[   +2 ms] [+1954 ms] /home/nghialm/.asdf/installs/flutter/2.0.2-stable/bin/cache/dart-sdk/bin/dart --disable-dart-dev /home/nghialm/.asdf/installs/flutter/2.0.2-stable/bin/cache/artifacts/engine/linux-x64/frontend_server.dart.snapshot --sdk-root /home/nghialm/.asdf/installs/flutter/2.0.2-stable/bin/cache/artifacts/engine/common/flutter_patched_sdk/ --target=flutter --no-print-incremental-dependencies -Ddart.vm.profile=false -Ddart.vm.product=false --enable-asserts --track-widget-creation --packages /home/nghialm/learn/flutter/bug/.dart_tool/package_config.json --output-dill /home/nghialm/learn/flutter/bug/.dart_tool/flutter_build/bf49fed71448292e8e50cf6c7160b3f9/app.dill --depfile /home/nghialm/learn/flutter/bug/.dart_tool/flutter_build/bf49fed71448292e8e50cf6c7160b3f9/kernel_snapshot.d package:bug/main.dart
[        ] [+10592 ms] kernel_snapshot: Complete
[        ] [ +493 ms] debug_bundle_linux_assets: Starting due to {}
[        ] [ +196 ms] debug_bundle_linux_assets: Complete
[        ] [ +720 ms] Persisting file store
[        ] [   +8 ms] Done persisting file store
[        ] [   +7 ms] build succeeded.
[        ] [   +9 ms] "flutter assemble" took 14,321ms.
[        ] [   +4 ms] Running shutdown hooks
[        ] [        ] Shutdown hooks complete
[        ] [        ] exiting with code 0
[ +657 ms] [2/6] Building CXX object CMakeFiles/bug.dir/main.cc.o
[  +84 ms] [3/6] Building CXX object CMakeFiles/bug.dir/flutter/generated_plugin_registrant.cc.o
[  +62 ms] [4/6] Building CXX object CMakeFiles/bug.dir/my_application.cc.o
[ +192 ms] [5/6] Linking CXX executable intermediates_do_not_run/bug
[        ] [5/6] Install the project...
[   +8 ms] -- Install configuration: "Debug"
[        ] -- Installing: /home/nghialm/learn/flutter/bug/build/linux/debug/bundle/bug
[        ] -- Set runtime path of "/home/nghialm/learn/flutter/bug/build/linux/debug/bundle/bug" to "$ORIGIN/lib"
[        ] -- Installing: /home/nghialm/learn/flutter/bug/build/linux/debug/bundle/data/icudtl.dat
[   +1 ms] -- Installing: /home/nghialm/learn/flutter/bug/build/linux/debug/bundle/lib/libflutter_linux_gtk.so
[ +101 ms] -- Installing: /home/nghialm/learn/flutter/bug/build/linux/debug/bundle/data/flutter_assets
[        ] -- Installing: /home/nghialm/learn/flutter/bug/build/linux/debug/bundle/data/flutter_assets/AssetManifest.json
[        ] -- Installing: /home/nghialm/learn/flutter/bug/build/linux/debug/bundle/data/flutter_assets/NOTICES
[        ] -- Installing: /home/nghialm/learn/flutter/bug/build/linux/debug/bundle/data/flutter_assets/version.json
[        ] -- Installing: /home/nghialm/learn/flutter/bug/build/linux/debug/bundle/data/flutter_assets/fonts
[        ] -- Installing: /home/nghialm/learn/flutter/bug/build/linux/debug/bundle/data/flutter_assets/fonts/MaterialIcons-Regular.otf
[   +1 ms] -- Installing: /home/nghialm/learn/flutter/bug/build/linux/debug/bundle/data/flutter_assets/packages
[        ] -- Installing: /home/nghialm/learn/flutter/bug/build/linux/debug/bundle/data/flutter_assets/packages/cupertino_icons
[        ] -- Installing: /home/nghialm/learn/flutter/bug/build/linux/debug/bundle/data/flutter_assets/packages/cupertino_icons/assets
[        ] -- Installing: /home/nghialm/learn/flutter/bug/build/linux/debug/bundle/data/flutter_assets/packages/cupertino_icons/assets/CupertinoIcons.ttf
[        ] -- Installing: /home/nghialm/learn/flutter/bug/build/linux/debug/bundle/data/flutter_assets/FontManifest.json
[        ] -- Installing: /home/nghialm/learn/flutter/bug/build/linux/debug/bundle/data/flutter_assets/kernel_blob.bin
[  +45 ms] Building Linux application... (completed in 17.2s)
[ +213 ms] Observatory URL on device: http://127.0.0.1:41971/7YSBcaFBR1Y=/
[   +4 ms] Caching compiled dill
[  +77 ms] Connecting to service protocol: http://127.0.0.1:41971/7YSBcaFBR1Y=/
[  +60 ms] [ERROR:flutter/lib/ui/ui_dart_state.cc(186)] Unhandled Exception: type 'Null' is not a subtype of type 'Object' in type cast
[        ] #0      CastMap.remove (dart:_internal/cast.dart:280:48)
[        ] #1      SharedPreferencesLinux.remove (package:shared_preferences_linux/shared_preferences_linux.dart:92:17)
[        ] <asynchronous suspension>
[        ] #2      main (package:bug/main.dart:7:3)
[        ] <asynchronous suspension>
[  +99 ms] Launching a Dart Developer Service (DDS) instance at http://127.0.0.1:0, connecting to VM service at http://127.0.0.1:41971/7YSBcaFBR1Y=/.
[  +68 ms] DDS is listening at http://127.0.0.1:38075/kl2BUUcCz64=/.
[  +44 ms] Successfully connected to service protocol: http://127.0.0.1:41971/7YSBcaFBR1Y=/
[ +875 ms] DevFS: Creating new filesystem on the device (null)
[  +12 ms] DevFS: Created new filesystem on the device (file:///tmp/bugIHSZGY/bug/)
[   +3 ms] Updating assets
[  +74 ms] Syncing files to device Linux...
[   +1 ms] <- reset
[        ] Compiling dart to kernel with 0 updated files
[   +2 ms] <- recompile package:bug/main.dart 5023e9ee-9037-4ef8-96a8-6280dce88a33
[        ] <- 5023e9ee-9037-4ef8-96a8-6280dce88a33
[ +127 ms] Updating files.
[        ] DevFS: Sync finished
[        ] Syncing files to device Linux... (completed in 132ms)
[        ] Synced 0.0MB.
[        ] <- accept
[   +4 ms] Connected to _flutterView/0x55639291eef0.
[   +1 ms] Flutter run key commands.
[   +1 ms] r Hot reload. 🔥🔥🔥
[        ] R Hot restart.
[        ] h Repeat this help message.
[        ] d Detach (terminate "flutter run" but leave application running).
[        ] c Clear the screen
[        ] q Quit (terminate the application on the device).
[        ] An Observatory debugger and profiler on Linux is available at: http://127.0.0.1:38075/kl2BUUcCz64=/
[        ] 
           Flutter DevTools, a Flutter debugger and profiler, on Linux is available at: http://127.0.0.1:40871?uri=http%3A%2F%2F127.0.0.1%3A38075%2Fkl2BUUcCz64%3D%2F
[        ] 💪 Running with sound null safety 💪
[  +88 ms] ** (bug:114000): WARNING **: 15:53:09.844: Unable to retrieve framework response: Message is not valid JSON
[ +127 ms] Service protocol connection closed.
[        ] Lost connection to device.
[   +2 ms] DevFS: Deleting filesystem on the device (file:///tmp/bugIHSZGY/bug/)
[ +255 ms] Ignored error while cleaning up DevFS: TimeoutException after 0:00:00.250000: Future not completed
[   +6 ms] "flutter run" took 19,797ms.
[   +9 ms] Running shutdown hooks
[        ] Shutdown hook priority 4
[   +7 ms] Shutdown hooks complete
[        ] exiting with code 0
```

```
Analyzing bug...
No issues found! (ran in 1.3s)
```

```
[✓] Flutter (Channel stable, 2.0.2, on Linux, locale en_US.UTF-8)
    • Flutter version 2.0.2 at /home/nghialm/.asdf/installs/flutter/2.0.2-stable
    • Framework revision 8962f6dc68 (6 days ago), 2021-03-11 13:22:20 -0800
    • Engine revision 5d8bf811b3
    • Dart version 2.12.1

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /opt/android-sdk
    • Platform android-30, build-tools 29.0.3
    • ANDROID_HOME = /opt/android-sdk
    • ANDROID_SDK_ROOT = /opt/android-sdk
    • Java binary at: /usr/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_282-b08)
    • All Android licenses accepted.

[✗] Chrome - develop for the web (Cannot find Chrome executable at google-chrome)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[✓] Linux toolchain - develop for Linux desktop
    • clang version 11.1.0
    • cmake version 3.19.6
    • ninja version 1.10.2
    • pkg-config version 1.7.3

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/linux#android-setup for detailed instructions).

[✓] Connected device (1 available)
    • Linux (desktop) • linux • linux-x64 • Linux

! Doctor found issues in 2 categories.
```

</details>

**Comment by darshankawar** [comment]
Tagging @markusaksli-nc to verify this on Linux.

**Comment by martin-robert-fink** [comment]
I have the same issue on Windows.  I have to do a `containsKey('key')` before attempting a remove operation.

**Comment by markusaksli-nc** [comment]
Ran into some issues on linux but was able to reproduce this on Windows with 
```yaml
environment:
  sdk: ">=2.12.0 <3.0.0"

dependencies:
  flutter:
    sdk: flutter
  shared_preferences: ^2.0.5
```
on the latest master `2.1.0-13.0.pre.175`.
<details>
 <summary>flutter doctor -v</summary>

```console
[√] Flutter (Channel master, 2.1.0-13.0.pre.175, on Microsoft Windows [Version 10.0.19041.867], locale et-EE)
    • Flutter version 2.1.0-13.0.pre.175 at C:\Development\flutter_master
    • Framework revision efbde443d2 (14 hours ago), 2021-03-18 13:40:08 -0700
    • Engine revision c4628b709c
    • Dart version 2.13.0 (build 2.13.0-149.0.dev)

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at C:\Users\marku\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: C:\Users\marku\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\201.7199119\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.7.7)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.7.30621.155
    • Windows 10 SDK version 10.0.18362.0

[√] Android Studio (version 4.1.0)
    • Android Studio at C:\Users\marku\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\201.7199119
    • Flutter plugin version 54.0.1
    • Dart plugin version 201.9335
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] Connected device (4 available)
    • SM G950F (mobile) • ce12171c51cc001c03 • android-arm64  • Android 9 (API 28)
    • Windows (desktop) • windows            • windows-x64    • Microsoft Windows [Version 10.0.19041.867]
    • Chrome (web)      • chrome             • web-javascript • Google Chrome 89.0.4389.90
    • Edge (web)        • edge               • web-javascript • Microsoft Edge 89.0.774.54

• No issues found!
```
</details>

**Comment by gspencergoog** [comment]
Is this being run with `--no-sound-null-safety`?

cc @stuartmorgan

**Comment by martin-robert-fink** [comment]
Nope - null safety is on.

**Comment by stuartmorgan** [comment]
This is https://github.com/dart-lang/sdk/issues/45473

We can work around it in the plugin by adding a `contains` check, but we shouldn't need to, so I'd like to see how quickly a fix could be turned around on the Dart side, and if we could cherry-pick it.

**Comment by nghialm269** [comment]
The issue was fixed in [2.2.1](https://github.com/flutter/flutter/wiki/Hotfixes-to-the-Stable-Channel#221-may-27-2021).
